test/integration/rmi_out_test.py

```diff
@@ -207,13 +226,34 @@ def test_consistency_of_data_stages(
             EXPECTED_ERRORS_DIR
             / f"expected_errors_{data_col}_{by_name}_{df1_name}_vs_{df2_name}.csv"
         )
+        if request.config.getoption("--five-year-coverage"):
+            expected_errors_path = (
+                EXPECTED_ERRORS_DIR / f"{expected_errors_path.stem}_five_year.csv"
+            )
         expected_aggregation_errors = pd.read_csv(expected_errors_path).set_index(by)
+        idx = actual_aggregation_errors.index
+        levels_set_types = [
+            "report_year",
+            "data_source",
+            "utility_id_pudl",
+        ]
+        if idx.names == levels_set_types:
+            actual_aggregation_errors.index = (
+                actual_aggregation_errors.index.set_levels(
+                    [
+                        idx.levels[0].astype(int),
+                        idx.levels[1].astype(str),
+                        idx.levels[2].astype(int),
+                    ],
+                    level=levels_set_types,
+                )
+            )
```

Comment:
This whole index level data type setting business is super janky but it fixes the weird one off error I was getting with the type mismatch. Maybe I should look into a more general/lasting solution here.

Reply:
yeah this looks janky/hard to understand what is happening.. if you are going to keep it i'd suggest at the least adding some why/what comments in here